sdk/core/Azure.Core.TestFramework/src/TestAppContext.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+
+namespace Azure.Core.TestFramework
+{
+    public class TestAppContext : DisposableConfig
+    {
+        private static SemaphoreSlim _lock { get; } = new(1, 1);
+        public TestAppContext(string name, string value) : base(name, value, _lock) { }
+        public TestAppContext(Dictionary<string, string> values) : base(values, _lock) { }
+        private MethodInfo removeMethod;
+        private object _switches;
+
+        internal override void SetValue(string name, string value)
+        {
+            if (!bool.TryParse(value, out bool boolVal))
+            {
+                throw new ArgumentException("Value must be parsable as bool");
+            }
+            if (AppContext.TryGetSwitch(name, out bool originalValue))
+            {
+                _originalValues[name] = originalValue.ToString();
+            }
+            else
+            {
+                _originalValues[name] = null;
+            }
+            AppContext.SetSwitch(name, boolVal);
+        }
+
+        internal override void SetValues(Dictionary<string, string> values)
+        {
+            foreach (var kvp in values)
+            {
+                SetValue(kvp.Key, kvp.Value);
+            }
+        }
+
+        internal override void InitValues()
+        {
+            Type type = typeof(AppContext);
+            FieldInfo info = type.GetField("s_switches", BindingFlags.NonPublic | BindingFlags.Static);
+            if (info == null)
+            {
+                info = type.GetField("s_switchMap", BindingFlags.NonPublic | BindingFlags.Static);
+            }
+            _switches = info.GetValue(null);
+            if (_switches == null)
```

Comment:
Do this unconditionally?

Reply:
I understand that. I'm saying we can simplify code by always setting a well-known switch.